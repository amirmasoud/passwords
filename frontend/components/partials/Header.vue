<template>
  <header
    class="flex items-center fixed w-full h-16 bg-white sm:flex sm:justify-between sm:items-center border-b border-gray-300"
  >
    <div class="hidden md:block md:w-64 md:px-2 md:py-2 w-full bg-gray-900">
      <div>
        <h1 class="text-2xl font-bold text-white">
          <key-svg class="inline mr-2 text-indigo-400" />
          <span class="text-indigo-400">terra</span>cotta
        </h1>
      </div>
    </div>
    <nav class="w-full md:-ml-64">
      <div class="flex items-center ml-2 md:ml-64">
        <button
          type="button"
          @click="toggle"
          class="block md:hidden text-gray-400 sm:px-2 hover:text-gray-600"
        >
          <menu-svg v-if="!isOpen" class="w-6" />
          <close-svg
            v-else
            class="w-6 p-1 relative z-10 bg-red-700 text-white rounded-full"
          />
        </button>
        <div
          class="w-full mr-2 relative max-w-screen-lg uppercase tracking-wide text-gray-500 text-xs font-bold appearance-none border-none text-gray-200 mx-auto md:pr-3"
        >
          <input
            type="search"
            class="w-full ml-2 md:ml-8 py-2 pl-2 pr-8 bg-gray-200 text-gray-500 border border-transparent focus:bg-white focus:border-gray-300 tracking-wide text-xs font-bold rounded outline-none focus:shadow-outline"
            placeholder="SEARCH"
          />
          <button
            class="flex items-center h-full absolute right-0 top-0 px-2 -mr-2 outline-none focus:shadow-outline"
          >
            <search-svg class="text-gray-400" />
          </button>
        </div>
        <div class="mx-2">
          <AccountDropdown class="sm:px-2" />
        </div>
      </div>
    </nav>
  </header>
</template>

<script>
import AccountDropdown from '~/components/elements/AccountDropdown'
import KeySvg from '~/components/svg/Key'
import SearchSvg from '~/components/svg/Search'
import MenuSvg from '~/components/svg/Menu'
import CloseSvg from '~/components/svg/Close'

export default {
  components: {
    AccountDropdown,
    'key-svg': KeySvg,
    'search-svg': SearchSvg,
    'menu-svg': MenuSvg,
    'close-svg': CloseSvg
  },
  data() {
    return {
      isOpen: false
    }
  },
  created() {
    this.$root.$on('close-menu', () => {
      this.isOpen = false
    })
  },
  methods: {
    toggle() {
      this.isOpen = !this.isOpen
      this.$root.$emit('menu-is-open', this.isOpen)
    }
  }
}
</script>
