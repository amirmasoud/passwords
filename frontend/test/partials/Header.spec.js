import { mount } from '@vue/test-utils'
import Header from '@/components/partials/Header'

describe('Header', () => {
  const wrapper = mount(Header)

  it('is a Vue instance', () => {
    expect(wrapper).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('button toggles the menu', async () => {
    const button = wrapper.find('button')
    await button.trigger('click')
    expect(wrapper.vm.isOpen).toBe(true)
    await button.trigger('click')
    expect(wrapper.vm.isOpen).toBe(false)
  })

  it('listens to close-menu custom event', async () => {
    wrapper.vm.$root.$emit('close-menu')
    await wrapper.vm.$nextTick()

    expect(wrapper.emitted('close-menu'))
    expect(wrapper.vm.isOpen).toBe(false)
  })
})
