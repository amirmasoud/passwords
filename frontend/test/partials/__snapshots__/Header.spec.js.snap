// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="flex items-center fixed w-full h-16 bg-white sm:flex sm:justify-between sm:items-center border-b border-gray-300"
>
  <div
    class="hidden md:block md:w-64 md:px-2 md:py-2 w-full bg-gray-900"
  >
    <div>
      <h1
        class="text-2xl font-bold text-white"
      >
        <svg
          class="inline mr-2 text-indigo-400"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="heroicon-ui"
            d="M11.85 17.56a1.5 1.5 0 0 1-1.06.44H10v.5c0 .83-.67 1.5-1.5 1.5H8v.5c0 .83-.67 1.5-1.5 1.5H4a2 2 0 0 1-2-2v-2.59A2 2 0 0 1 2.59 16l5.56-5.56A7.03 7.03 0 0 1 15 2a7 7 0 1 1-1.44 13.85l-1.7 1.71zm1.12-3.95l.58.18a5 5 0 1 0-3.34-3.34l.18.58L4 17.4V20h2v-.5c0-.83.67-1.5 1.5-1.5H8v-.5c0-.83.67-1.5 1.5-1.5h1.09l2.38-2.39zM18 9a1 1 0 0 1-2 0 1 1 0 0 0-1-1 1 1 0 0 1 0-2 3 3 0 0 1 3 3z"
          />
        </svg>
         
        <span
          class="text-indigo-400"
        >
          terra
        </span>
        cotta
      
      </h1>
    </div>
  </div>
   
  <nav
    class="w-full md:-ml-64"
  >
    <div
      class="flex items-center ml-2 md:ml-64"
    >
      <button
        class="block md:hidden text-gray-400 sm:px-2 hover:text-gray-600"
        type="button"
      >
        <svg
          class="w-6"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="heroicon-ui"
            d="M4 5h16a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2z"
          />
        </svg>
      </button>
       
      <div
        class="w-full mr-2 relative max-w-screen-lg uppercase tracking-wide text-gray-500 text-xs font-bold appearance-none border-none text-gray-200 mx-auto md:pr-3"
      >
        <input
          class="w-full ml-2 md:ml-8 py-2 pl-2 pr-8 bg-gray-200 text-gray-500 border border-transparent focus:bg-white focus:border-gray-300 tracking-wide text-xs font-bold rounded outline-none focus:shadow-outline"
          placeholder="SEARCH"
          type="search"
        />
         
        <button
          class="flex items-center h-full absolute right-0 top-0 px-2 -mr-2 outline-none focus:shadow-outline"
        >
          <svg
            class="text-gray-400"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="heroicon-ui"
              d="M16.32 14.9l5.39 5.4a1 1 0 0 1-1.42 1.4l-5.38-5.38a8 8 0 1 1 1.41-1.41zM10 16a6 6 0 1 0 0-12 6 6 0 0 0 0 12z"
            />
          </svg>
        </button>
      </div>
       
      <div
        class="mx-2"
      >
        <div
          class="relative sm:px-2"
        >
          <button
            class="account-btn relative block h-8 w-8 rounded-full overflow-hidden border-2 border-gray-300 focus:outline-none focus:border-white z-0"
          >
            <img
              alt="Your avatar"
              class="h-full w-full object-cover"
              src="https://images.unsplash.com/photo-1531295695962-cb83d1692a25?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=256&q=80"
            />
          </button>
           
          <!---->
           
          <!---->
        </div>
      </div>
    </div>
  </nav>
</header>
`;
